pkg/controller/seed-controller-manager/kubernetes/health.go

```diff
@@ -43,7 +43,6 @@ func (r *Reconciler) clusterHealth(ctx context.Context, cluster *kubermaticv1.Cl
 		resources.ApiserverDeploymentName:             {healthStatus: &extendedHealth.Apiserver, minReady: 1},
 		resources.ControllerManagerDeploymentName:     {healthStatus: &extendedHealth.Controller, minReady: 1},
 		resources.SchedulerDeploymentName:             {healthStatus: &extendedHealth.Scheduler, minReady: 1},
-		resources.MachineControllerDeploymentName:     {healthStatus: &extendedHealth.MachineController, minReady: 1},
```

Comment:
Did you test this? I mean it could cause some issues to the UI but I am not quite sure tbh. As this being checked in the seed and moved to the user cluster controller manager, the status of the machine controller in the UI might be fiddled with.

Reply:
Also I am not quite sure that this should really be moved to the user cluster controller manager. Why not simply check in initialMDController in the seed, the health check of the machien-controller-webhook, and then apply the initial MD, wouldn't this be more straightforward?

Reply:
Here, https://github.com/kubermatic/kubermatic/blob/master/pkg/apis/kubermatic/v1/cluster.go#L962

Reply:
It is not enough to check only the machineController and machineControllerWebhook deployment healths, we should check the existence of the mutatingWebhookConfiguration as well. It is deployed in the user cluster and not easily retrievable from that part of the seed-controller-manager. I moved the check to the user-cluster-controller-manager because here we have the certainty the configuration has been correctly created (the machineController health update means that all the resources have been correctly created). Otherwise, we could fall in having both the deployments created and healthy, but the configuration does not exist yet (and the validation won't be performed).

Reply:
And we are following the same approach that we use for MLA and OPA: https://github.com/kubermatic/kubermatic/blob/3c750f28c78acbc80c16493536e86033a7038c6b/pkg/controller/user-cluster-controller-manager/resources/reconciler.go#L1083-L1117
if the controller deploys a resource, it's up to it to update the cluster health for what concerns that specific resource (and the MC is deployed by the user-cluster-controller-manager).

Reply:
> It is not enough to check only the machineController and machineControllerWebhook deployment healths, we should check the existence of the mutatingWebhookConfiguration as well.

As I agree with you with this, I didn't only mean checking the health checks, what I meant is, you could actually achieve whatever you are trying to achieve in that controller as well. 

> It is deployed in the user cluster and not easily retrievable from that part of the seed-controller-manager. 

However, I don't agree with this point here, you can get whatever you like from that user cluster, using that controller as you can see over here https://github.com/kubermatic/kubermatic/blob/master/pkg/controller/seed-controller-manager/initialmachinedeployment/controller.go#L149. 

Here you have the user cluster client which you can use to get whatever resource in that user cluster. The difference between this, OPA and MLA is, those don't have a prior and dedicated controller so people simply throw it in there, but if you have a dedicated controller for what you are validating, just extend the health check over there instead of. moving a functionality from its right place.    

> if the controller deploys a resource, it's up to it to update the cluster health for what concerns that specific resource (and the MC is deployed by the user-cluster-controller-manager).

That's actually not true, MC is deployed via the seed controller manager not the user cluster controller manager: https://github.com/kubermatic/kubermatic/blob/master/pkg/controller/seed-controller-manager/kubernetes/resources.go#L298-L299

Reply:
You are right, it is deployed by the seed controller manager. I'll move this check there